### PR TITLE
Use ClimaReproducibilityTests.jl

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR replaces NCRegressionTests with ClimaReproducibilityTests.jl, which doesn't depend on NCDatasets. This interface is more transparent, has a path to lighter dependencies, and better matches the name of what we're doing.

### CODE REVIEW

File: test/Project.toml [hunk 1]
```diff
@@ -9,6 +9,7 @@ ClimaAtmos = "b2c96348-7fb7-4fe0-8da9-78d88439e717"
 ClimaComms = "3a4d1b5c-c61d-41fd-a00a-5873ba7a1b0d"
 ClimaCore = "d414da3d-4745-48bb-8d80-42e94e092884"
 ClimaParams = "5c42b081-d73a-476f-9059-fd94b934656c"
+ClimaReproducibilityTests = "e0c89595-00ba-42a9-9f9b-061ef3dc23a1"
```
Comment (Sbozzolo) [inline test/Project.toml:12]: Is this used in the test environment?

Comment (charleskawczynski) [reply test/Project.toml:12]: Good question, I'll take a look.

Comment (charleskawczynski) [reply test/Project.toml:12]: I don't think so, I'll try removing it.

**Review by nefrathenrici — APPROVED** [review]
Thanks Charlie!

**Pull request merged**